pkg/system/phase4_configuring.go

```diff
@@ -1089,3 +1098,133 @@ func (r *Reconciler) ReconcileDeploymentEndpointStatus() error {
+				nsStore.Spec.Type = nbv1.NSStoreTypeAzureBlob
+				secret.StringData["AccountName"] = nsr.Identity
+				secret.StringData["AccountKey"] = namespaceResourceOperatorInfo.SecretKey
+				secret.Name = fmt.Sprintf("namespace-store-%s-%s", nbv1.NSStoreTypeAzureBlob, nsr.Name)
+				nsStore.Spec.AzureBlob = &nbv1.AzureBlobSpec{
+					TargetBlobContainer: nsr.TargetBucket,
+					Secret: corev1.SecretReference{
+						Name:      secret.Name,
+						Namespace: secret.Namespace,
+					},
+				}
+			case "S3_COMPATIBLE":
+				nsStore.Spec.Type = nbv1.NSStoreTypeAzureBlob
+				secret.Name = fmt.Sprintf("namespace-store-%s-%s", nbv1.NSStoreTypeS3Compatible, nsr.Name)
+				secret.StringData["AWS_ACCESS_KEY_ID"] = nsr.Identity
+				secret.StringData["AWS_SECRET_ACCESS_KEY"] = namespaceResourceOperatorInfo.SecretKey
+
+				var sig nbv1.S3SignatureVersion
+				if nsr.AuthMethod == nb.CloudAuthMethodAwsV4 {
+					sig = nbv1.S3SignatureVersionV4
+				}
+				if nsr.AuthMethod == nb.CloudAuthMethodAwsV2 {
+					sig = nbv1.S3SignatureVersionV2
+				}
+				nsStore.Spec.S3Compatible = &nbv1.S3CompatibleSpec{
+					TargetBucket: nsr.TargetBucket,
+					Endpoint:     nsr.Endpoint,
+					Secret: corev1.SecretReference{
+						Name:      secret.Name,
+						Namespace: secret.Namespace,
+					},
+					SignatureVersion: sig,
+				}
+			case "IBM_COS":
+				nsStore.Spec.Type = nbv1.NSStoreTypeAzureBlob
+				secret.Name = fmt.Sprintf("namespace-store-%s-%s", nbv1.NSStoreTypeIBMCos, nsr.Name)
+				secret.StringData["AWS_ACCESS_KEY_ID"] = nsr.Identity
+				secret.StringData["AWS_SECRET_ACCESS_KEY"] = namespaceResourceOperatorInfo.SecretKey
+
+				var sig nbv1.S3SignatureVersion
+				if nsr.AuthMethod == nb.CloudAuthMethodAwsV4 {
+					sig = nbv1.S3SignatureVersionV4
+				}
+				if nsr.AuthMethod == nb.CloudAuthMethodAwsV2 {
+					sig = nbv1.S3SignatureVersionV2
+				}
+				nsStore.Spec.IBMCos = &nbv1.IBMCosSpec{
+					TargetBucket: nsr.TargetBucket,
+					Endpoint:     nsr.Endpoint,
+					Secret: corev1.SecretReference{
+						Name:      secret.Name,
+						Namespace: secret.Namespace,
+					},
+					SignatureVersion: sig,
+				}
+			default:
+				logrus.Errorf(`❌ Could not create NamespaceStore %q invalid endpoint type %q`, nsStore.Name, nsr.EndpointType)
+				continue
+			}
+
+			// Create namespace store CR
+			util.Panic(controllerutil.SetControllerReference(r.NooBaa, nsStore, scheme.Scheme))
+			if !util.KubeCreateSkipExisting(nsStore) {
+				logrus.Errorf(`❌ Could not create NamespaceStore %q in Namespace %q (conflict)`, nsStore.Name, nsStore.Namespace)
+				continue
+			}
+
+			if !util.KubeCheck(secret) {
+				// Create secret
+				util.Panic(controllerutil.SetControllerReference(nsStore, secret, scheme.Scheme))
+				if !util.KubeCreateSkipExisting(secret) {
+					logrus.Errorf(`❌ Could not create Secret %q in Namespace %q (conflict)`, secret.Name, secret.Namespace)
+					continue
+				}
+			}
+			err = r.NBClient.SetNamespaceStoreInfo(nb.NamespaceStoreInfo{
```

Comment:
not sure I understand the flow. once you create a NS store wouldn't its reconciler try to create a NS resource in noobaa? maybe I missed it, but I didn't see where it's handled

Reply:
no it shouldn't, we are talking about upgrade so it already exist as namespace resource before the upgrade.
when calling read system it already exist so it won't be called..

Reply:
👍